lib/jobs/provision_proxies.rb

```diff
@@ -1,13 +1,16 @@
 module Jobs
   class ProvisionProxies
     class << self
-      def perform(site_id, source_id, num_proxies)
+      def perform(site_id, source_id, desired_proxy_count)
         source = Source.find source_id
-        proxies = source.provision_proxies num_proxies
+        num_proxies = desired_proxy_count - source.proxies.length
+        if num_proxies > 0
+          proxies = source.provision_proxies num_proxies
 
-        site = Site.find site_id
-        site.proxies.concat(*proxies)
-        site.save
+          site = Site.find site_id
+          site.proxies.concat(*proxies)
```

Comment:
Good catch!  The only reason I made a change to this file was to protect against creating 100 proxies when we get 10 requests all at once to create 10 proxies.  Turns out it also needs to call the new method I created.